Canvas/CanvasUITests/Profile/ProfileTests.swift

```diff
@@ -18,27 +18,56 @@ import XCTest
 import TestsFoundation
 
 class ProfileTests: CanvasUITests {
+
+    func testCourseCardGrades() {
+        Profile.open()
+        Profile.showGradesToggle.waitToExist()
+        if !Profile.showGradesToggle.isSelected {
+            Profile.showGradesToggle.tap()
+        }
+        Profile.close()
+        Dashboard.courseCard(id: "263").waitToExist()
+        XCTAssertEqual(Dashboard.courseCard(id: "263").label, "Assignments 70%")
+
+        Profile.open()
+        Profile.showGradesToggle.tap()
+        Profile.close()
+        Dashboard.courseCard(id: "263").waitToExist()
+        XCTAssertEqual(Dashboard.courseCard(id: "263").label.trimmingCharacters(in: .whitespacesAndNewlines), "Assignments")
+    }
+
     func testProfileDisplaysUsername() {
         Dashboard.profileButton.waitToExist()
-        Dashboard.profileButton.tap()
+        Profile.open()
         Profile.userNameLabel.waitToExist()
         XCTAssertEqual(Profile.userNameLabel.label, "Student One")
     }
 
     func testProfileChangesUser() {
         Dashboard.profileButton.waitToExist()
-        Dashboard.profileButton.tap()
+        Profile.open()
         Profile.changeUserButton.tap()
         let entry = user!.keychainEntry!
         LoginStartKeychainEntry.cell(host: entry.baseURL.host!, userID: entry.userID).waitToExist()
     }
 
     func testProfileLogsOut() {
         Dashboard.profileButton.waitToExist()
-        Dashboard.profileButton.tap()
+        Profile.open()
         Profile.logOutButton.tap()
         LoginStart.findSchoolButton.waitToExist()
         let entry = user!.keychainEntry!
         XCTAssertFalse(LoginStartKeychainEntry.cell(host: entry.baseURL.host!, userID: entry.userID).exists)
     }
+
+    func testPreviewUserFile() {
+        Profile.open()
+        Profile.filesButton.tap()
+
+        // Files List
+        FilesList.file(id: "11585").waitToExist()
+        FilesList.file(id: "11585").tap()
+        sleep(3)
```

Comment:
Make sense.. give it time to download the file.

Reply:
maybe that would be a useful comment. `// allow time to download the file`